src/Storages/System/StorageSystemFunctions.cpp

```diff
@@ -102,6 +118,7 @@ NamesAndTypesList StorageSystemFunctions::getNamesAndTypes()
     return {
         {"name", std::make_shared<DataTypeString>()},
         {"is_aggregate", std::make_shared<DataTypeUInt8>()},
+        {"is_deterministic", std::make_shared<DataTypeNullable>(std::make_shared<DataTypeUInt8>())},
```

Comment:
why is it nullable?

Reply:
See l. 50 and l. 141ff. There are situations in which we cannot determine if a function is deterministic or not. In this case, NULL is displayed.